daemon/health.go

```diff
@@ -0,0 +1,312 @@
+package daemon
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types/backend"
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/daemon/exec"
+	"github.com/docker/engine-api/types/strslice"
+)
+
+const (
+	// Longest healthcheck probe output message to store. Longer messages will be truncated.
+	maxOutputLen = 255
+
+	// Default interval between probe runs (from the end of the first to the start of the second).
+	// Also the time before the first probe.
+	defaultProbeInterval = 30 * time.Second
+
+	// The maximum length of time a single probe run should take. If the probe takes longer
+	// than this, the check is considered to have failed.
+	defaultProbeTimeout = 30 * time.Second
+
+	// Probes started within this period do not cause transitions from Starting to Unhealthy.
+	// Instead, we remain in Starting in this case (however, we can still move to Unhealthy after
+	// a successful probe, even if we're still in the grace period).
+	defaultGracePeriod = 30 * time.Second
+
+	// Shut down a container if it becomes Unhealthy.
+	defaultExitOnUnhealthy = true
```

Comment:
@aluzzardi Could you comment here? I think you asked for the default to be `true`.